src/resolve/resolved-dns-query.c

```diff
@@ -381,6 +381,8 @@ DnsQuery *dns_query_free(DnsQuery *q) {
         if (!q)
                 return NULL;
 
+        sd_event_source_disable_unref(q->timeout_event_source);
```

Comment:
i'd probably write this as:

```c
q->timeout_event_source = sd_event_source_disable_unref(q->timeout_event_source);
```

i.e. we call into dns_query_free(), and all kinds of other stuff, which might refer back to us and might be tripped up because we field is left pointing to freed data. i mean, i am not saying there's a bug *now*, I am just saying that after some later refactoring where dns_query_free() does more stuff it might become one and we should avoid that right away.

i.e. i think we can get away with leaving bad pointers around during destruction of some object, when we know we are really the only ones referencing the object. But here it's not that clear I think... the aux queries might still have a pointer back to us...

Reply:
OK. Actually `DnsQuery` is a complicated object.